### PR TITLE
[Core] make remove installation directories optional

### PR DESCRIPTION
**Pull request by jrubiogonzalez**
**📝 Description**
This PR makes removing KratosMultiphysics and libs directory optional. This might be dangerous if some applications get stuck, but on our case as we have plenty of external dll's and several applications this reduces the build time for small changes significantly.

### CODE REVIEW

**Review by roigcarlo — DISMISSED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +46,12 @@ if(NOT DEFINED INSTALL_RUNKRATOS)
   set(INSTALL_RUNKRATOS ON)
 endif(NOT DEFINED INSTALL_RUNKRATOS)
 
+if(NOT DEFINED REMOVE_INSTALL_DIRECTORIES)
+  message("-- No REMOVE_INSTALL_DIRECTORIES is defined, setting to ON")
+  set(REMOVE_INSTALL_DIRECTORIES ON)
+endif(NOT DEFINED REMOVE_INSTALL_DIRECTORIES)
```
Comment (philbucher) [inline CMakeLists.txt:52]: option(REMOVE_INSTALL_DIRECTORIES "Remove the install directories to achieve a clean compilation" ON)

**Review by philbucher — CHANGES_REQUESTED** [review]
please use more modern CMake syntax

File: CMakeLists.txt (hunk 1 above)
Comment (philbucher) [inline CMakeLists.txt]:
```suggestion
option(REMOVE_INSTALL_DIRECTORIES "Remove the install directories to achieve a clean compilation" ON)
```

please use modern Cmake Syntax

**Comment by matekelemen** [comment]
@jrubiogonzalez just out of curiosity:
- why does it take so much time to copy the external libs back into the install dir?
   - are you building them directly into the install dir?
   - or do you have to manually copy them there?
- how many/large external libs and apps are we talking about?
- can you give me a rough estimate of how much faster the build process is without clearing the install dir first?

**Comment by jrubiogonzalez** [comment]
> 

We are compiling around 17 applications total between public and private and adding few extra libraries, including mkl. That sums up about 700Mb in win and far more in linux.

If we add our environment anti-virus that is quite intrusive (specially in windows) we go in a 0 build (no changes at all) in my workstation from 15s to 5s.

**Review by philbucher — APPROVED** [review]
Nice, thx!

File: CMakeLists.txt (hunk 1 above)
Comment (roigcarlo) [reply CMakeLists.txt]: We should definitely change that all over kratos...

**Pull request merged**